openmls/src/group/mls_group/proposal.rs

```diff
@@ -361,14 +392,14 @@ impl MlsGroup {
     /// Returns an error when the group does not support all the required capabilities
     /// in the new `extensions`.
     #[allow(clippy::type_complexity)]
-    pub fn update_group_context_extensions<KeyStore: OpenMlsKeyStore>(
+    pub fn update_group_context_extensions<Provider: OpenMlsProvider>(
         &mut self,
-        provider: &impl OpenMlsProvider<KeyStoreProvider = KeyStore>,
+        provider: &impl OpenMlsProvider,
```

Comment:
```suggestion
        provider: &Provider,
```

If you replace this with Provider, the Error conversion will work and we can replace the `unwrap()` below with a `?`